### ISSUE TITLE
[HOLD for payment 2024-05-06] [HOLD for payment 2024-05-03] CRITICAL: [Actionable Whispers] [$500] Workspace list doesn't show up after creating a workspace for categorize it

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#40406]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.62-13
**Reproducible in staging?:** Yes
**Reproducible in production?:** No, new feature
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Issue reported by:** Applause-Internal team

## Action Performed:
1. Navigate to staging.new.expensify.com
2. Create a new account
3. Go to your space
4. Create a manual track expense
5. Click on Categorize it
6. Click on Learn more and create a new workspace
7. Go to your space and click on Categorize it again

## Expected Result:
The newly created workspace should show up

## Actual Result:
No workspace found shows up

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/Expensify/App/assets/115492554/53149281-37f9-481f-90dd-760d98149cc0

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~012e058847a79bee4c</li>
        <li>Upwork Job ID: 1781103150081294336</li>
        <li>Last Price Increase: 2024-04-19</li>
<li>Automatic offers: </li>
<ul>
<li>c3024 | Reviewer | 0</li>
<li>tienifr | Contributor | 0</li>
</ul></li>
    </ul>
</details>

**Comment by izarutskaya** [comment]
@puneetlath I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors.

**Comment by puneetlath** [comment]
@shubham1206agra I don't totally follow. Are you saying this is expected and not a deploy blocker?

**Comment by thienlnam** [comment]
I don't think this needs to block deploy, it is a new feature - I think this only happens with new workspaces and is only sometimes reproducible. We can probably make this external

**Comment by thienlnam** [comment]
I can also take this over since it's related to track - but feel free to re-assign if you're interested

**Comment by c3024** [comment]
RCA suggested by @tienifr [here](https://github.com/Expensify/App/issues/40406#issuecomment-2065633395) makes sense. The suggested solution looks good and works well.

🎀 👀 🎀 C+ Reviewed

**Comment by quinthar** [comment]
Great!  @c3024 can you give an ETA on this?

**Comment by c3024** [comment]
@tienifr 

Could you raise the PR today? This is a critical issue. ^^

**Comment by c3024** [comment]
@tienifr can you give an ETA for opening the PR?

**Comment by c3024** [comment]
@thienlnam 

We have here maxRecentReportsToShow as 5

https://github.com/Expensify/App/blob/690609052072c89c52f24b21d95ea02b2819a769/src/libs/OptionsListUtils.ts#L2011

I can change this if needed.

**Comment by thienlnam** [comment]
Yup please do, there should be no limit

**Comment by thienlnam** [comment]
However, why didn't the report show up during search then?

**Comment by c3024** [comment]
That is what I am trying to find as well.

**Comment by c3024** [comment]
It looks like this here 
https://github.com/Expensify/App/blob/690609052072c89c52f24b21d95ea02b2819a769/src/pages/workspace/WorkspacesListPage.tsx#L379
creates a draft but does not write to backend.

Search after Logging out and logging in works fine even with maxRecentReportsToShow as 5.

**Comment by puneetlath** [comment]
@c3024 bump on the checklist. 

@tienifr has been paid.

**Comment by puneetlath** [comment]
Issue for regression test: https://github.com/Expensify/Expensify/issues/393986

Everyone has been paid. Thanks!

**Comment by puneetlath** [comment]
Ah ok. So there were two PRs for this issue? @thienlnam can you confirm who all needs to be paid out here?

**Comment by thienlnam** [comment]
This PR fixes what was described in the issue description
https://github.com/Expensify/App/pull/40730

I thought this wasn't solved when I was testing it [here](https://github.com/Expensify/App/issues/40406#issuecomment-2077613212), but turns out there was a different problem with the limit of workspaces shown in the selector which was fixed https://github.com/Expensify/App/pull/41009

**Comment by thienlnam** [comment]
@c3024 worked on both PRs, so we could give them another payment for this [PR](https://github.com/Expensify/App/pull/41009), and have a payment to @ishpaul777 for being the C+ on that PR as well

**Comment by puneetlath** [comment]
Ok so it's: 
- $500 for @tienifr for https://github.com/Expensify/App/pull/40730 (already paid)
- $500 for @c3024 for https://github.com/Expensify/App/pull/40730 (already paid)
- $500 for @c3024 for https://github.com/Expensify/App/pull/41009
- $500 for @ishpaul777 for https://github.com/Expensify/App/pull/41009

Is that right?

**Comment by thienlnam** [comment]
Yup that's correct

**Comment by puneetlath** [comment]
Ok @tienifr and @c3024 are now fully paid. 

@ishpaul777 sent you an offer: https://www.upwork.com/nx/wm/offer/102304232

Ping me here when you've accepted please.
